tycho-core/src/main/java/org/eclipse/tycho/core/resolver/DefaultTychoResolver.java

```diff
@@ -138,10 +139,28 @@ public void resolveProject(MavenSession session, MavenProject project, List<Reac
 
         dr.setDependencyArtifacts(session, reactorProject, dependencyArtifacts);
 
+        DependencyArtifacts testDependencyArtifacts = dependencyArtifacts;
+        TychoProject tychoProjectType = projectTypes.get(project.getPackaging());
+        if (tychoProjectType instanceof BundleProject) {
+            /*
+             * TODO At the moment, we don't resolve test-specific deps, so we set it to common deps.
+             * This will be refined later with addition of support of test-specific deps from
+             * .classpath.
+             */
+            List<Dependency> testDependencies = ((BundleProject) tychoProjectType).getExtraTestRequirements();
+            if (!testDependencies.isEmpty()) {
+                logger.info("Resolving test dependencies of " + project);
+                resolverConfiguration.getExtraRequirements().addAll(testDependencies);
+                testDependencyArtifacts = resolver.resolveDependencies(session, project, preliminaryTargetPlatform,
```

Comment:
How does P2 resolver knows about those things? at least in my previous testing the P2 resolver needed them to build the set of required Units or does it means it will resolve the deps twice?
Also as mentioned above one actually don't want to resolve a given unit but a given requirement that boils down to 

- namespace
- properties (e.g. versionrange, id and so on...)

Reply:
Not sure exactly what you mean by "those things".
p2 resolver does translate extra requirements (as org.apache.maven.model.Dependency) into p2 requirements and run against them.